File: 56-merge-intervals/merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        intervals.sort(key=lambda x:x[0])
        res=[]
        res.append(intervals[0])
        for i in range(1,len(intervals)):
            top=res[-1]
            if intervals[i][0]<=top[1]:
                res.pop()
                y=intervals[i][1]
                if top[1]>y:
                    y=top[1]
                res.append([top[0],y])
            else:
                res.append(intervals[i])
        return res


        

File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        vector<vector<int>> ans;
        int n=intervals.size();
        ans.push_back(intervals[0]);
        for(int i=1;i<n;i++){
            auto a=ans.back();
            if(a[1]>=intervals[i][0]){
                ans.pop_back();
                ans.push_back({a[0],max(a[1],intervals[i][1])});
            }else{
                ans.push_back(intervals[i]);
            }
        }
        return ans;
    }
};